FLT/DedekindDomain/FiniteAdeleRing/IsDirectLimitRestricted.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2025 Madison Crim. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Madison Crim
+-/
+
+import FLT.Mathlib.Topology.Algebra.RestrictedProduct.Basic
+import FLT.DedekindDomain.FiniteAdeleRing.TensorPi
+import FLT.DedekindDomain.FiniteAdeleRing.IsDirectLimit
+
+section RestrictedProduct
+
+open Set Filter
+
+variable {ι : Type*} {𝓕 : Filter ι}
+
+variable (𝓕) in
+/-- The complements of sets in a `Filter`.
+E.g. for the cofinite filter, these are just the finite subsets. -/
+def Filter.complement : Set (Set ι) := (fun S => Sᶜ) '' 𝓕.sets
```

Comment:
I am slightly not sure if we want this or not. I am guessing we would prefer passing `hS : S ∉ 𝓕` around.

Reply:
Yes you're right, this isn't `S ∉ 𝓕`, this is `Sᶜ ∈ 𝓕`, it's a so-called ideal. We seem to have them in mathlib as `Order.Ideal` (defined on general preorders rather than just on `Set X` which is what you want here) but I can't see the link to filters.

Reply:
I am working on removing this definition and more generally this file from the PR.